core/arch/arm/kernel/thread_optee_smc.c

```diff
@@ -587,6 +600,9 @@ struct mobj *thread_rpc_alloc_payload(size_t size)
 
 struct mobj *thread_rpc_alloc_kernel_payload(size_t size)
 {
+	if (IS_ENABLED(CFG_CORE_DYN_SHM) && size > SMALL_PAGE_SIZE)
```

Comment:
Please add a comment explaining this check. Perhaps:
```
/*
 * Error out early since kernel private dynamic shared memory allocations
 * currently doesn't use the `OPTEE_MSG_ATTR_NONCONTIG` bit and cannot be
 * larger than a page.
 */

Reply:
ok. I had to reword a little since it was going beyond the 80 character line limit (not sure if that is an issue for OP-TEE comments? but just in case)